test/test_gpio.c

```diff
@@ -0,0 +1,217 @@
+#include "gpio.h"
+
+#include <stdint.h>
+
+#include "unity.h"
+
+void setup_test(void) {}
+
+void teardown_test(void) {}
+
+// gpio_init
+
+// Tests that the GPIO init works. This should simply return true 100% of time unless the test is on
+// x86 and the configuration is incorrect in which case fix it!
+void test_gpio_valid_init(void) { TEST_ASSERT_TRUE(gpio_init()); }
+
+// gpio_init_pin
+
+// Test that a valid gpio configuration will work.
+void test_gpio_valid_init_pin(void) {
+  // Default settings for a pin.
+  GPIOSettings settings = {.direction = GPIO_DIR_IN,
+                           .state = GPIO_STATE_LOW,
+                           .resistor = GPIO_RES_NONE,
+                           .alt_function = GPIO_ALTFN_NONE };
+  // A pin that should be valid on all configurations.
+  GPIOAddress address = {.port = 0, .pin = 0 };
+  TEST_ASSERT_TRUE(gpio_init());
```

Comment:
If you're calling this in every single test case, might as well move it to setup_test :)